app/views/layouts/application.html.erb

```diff
@@ -65,46 +66,55 @@
 
   <body <% if content_for :body_class %>class="<%= yield :body_class %>"<% elsif state_file? %>class="state-file"<% end %>>
     <div class="page-wrapper">
-      <div data-component="ClientMenuOverlay" class="menu-overlay"></div>
-      <% if state_file? %>
-        <%= render 'state_file/state_file_pages/client_menu' %>
-      <% elsif !hub? %>
-        <%= render "shared/client_menu" %>
-      <% end %>
+
       <div class="page-content">
-        <a href="#maincontent" id="skip-content-link" class="skip-link button--green"><%= t('views.layouts.application.skip_content')%></a>
-        <% if content_for :header %>
-          <%= yield :header %>
-        <% elsif hub? %>
-          <%= render 'shared/hub_header' %>
-        <% elsif state_file? %>
-          <%= render 'state_file/state_file_pages/header' %>
-        <% else %>
-          <%= render 'shared/header', variation: "gyr", home_link: root_path %>
-        <% end %>
+        <header role="banner">
+          <a href="#maincontent" id="skip-content-link" class="skip-link button--green"><%= t('views.layouts.application.skip_content')%></a>
+
+          <div data-component="ClientMenuOverlay" class="menu-overlay"></div>
+          <% if state_file? %>
+            <%= render 'state_file/state_file_pages/client_menu' %>
+          <% elsif !hub? %>
+            <%= render "shared/client_menu" %>
+          <% end %>
+
+          <% if content_for :header %>
+            <%= yield :header %>
+          <% elsif hub? %>
+            <%= render 'shared/hub_header' %>
+          <% elsif state_file? %>
+            <%= render 'state_file/state_file_pages/header' %>
+          <% else %>
+            <%= render 'shared/header', variation: "gyr", home_link: root_path %>
+          <% end %>
 
-        <%= render "shared/environment_warning" %>
+          <%= render "shared/environment_warning" %>
+
+          <% if content_for(:back_to) %>
+            <%= render "hub/back_to", back_to: content_for(:back_to)%>
+          <% end %>
 
-        <% if content_for(:back_to) %>
-          <%= render "hub/back_to", back_to: content_for(:back_to)%>
-        <% end %>
-        <div id="maincontent">
           <div class="flash-alerts">
             <%= render "shared/flash_alerts", flash: flash %>
           </div>
+
           <%= yield(:notice) if content_for?(:notice) %>
+        </header>
 
+        <main id="maincontent">
           <%= content_for?(:main) ? yield(:main) : yield %>
-          <% if content_for :footer %>
-            <%= yield :footer %>
-          <% elsif state_file? %>
-            <%= render 'state_file/state_file_pages/footer' %>
-          <% else %>
-            <%= render 'shared/footer' %>
-          <% end %>
-        </div>
+        </main>
+
+        <% if content_for :footer %>
```

Comment:
Looks like <header> and <footer> are defined in different places as of this change - should that be the case? Is there a benefit to bringing the <footer> element up into this layout, or is it impractical?

Reply:
TBH I was trying to make the lightest touch possible given the scope of the other changes I had to make, and it seemed like the footer didn't need intervention. Happy to bring it in here though if that's better!

Reply:
That's reasonable - just wanted to make sure I understood the rationale for the different treatment. I'm fine with it staying like this!